File: sort_algorithms/bubble_sort.py
# Bubble Sort
# Python
# Porter L

def bubbleSort(Arr): 
    length = len(Arr) 
    for x in range(length-1): 
        for y in range(0, length-x-1): 
            if (Arr[y] > Arr[y+1]): 
                Arr[y], Arr[y+1] = Arr[y+1], Arr[y] 
    return Arr

# Example
if __name__=='__main__': 
    test_array = [2,7,22,71,5,3,11,9,7,5,4,3,56,3,23,43,97,3]
    print(test_array) # unsorted
    test_array = bubbleSort(test_array)
    print(test_array) # sorted

File: sort_algorithms/javascript/bubble_sort.js
// Bubble Sort
// Javascript
// Porter L

function bubbleSort(Arr){
    console.log('bubble sort');
    var length = Arr.length;
    for (x = 0; x < length-1; x++){
        for (y = 0; y < length-x-1; y++){
            if (Arr[y] > Arr[y+1]){
                var b = Arr[y];
                Arr[y] = Arr[y+1];
                Arr[y+1] = b;
            }
        }
    }
    return Arr
}